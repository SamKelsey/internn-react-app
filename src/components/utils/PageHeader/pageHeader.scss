@import "../../styleVariables";

.pageHeader {
  margin: 10px 0;
  text-align: center;
  font-family: "Montserrat", sans-serif;

  h1 {
    display: inline-block;
    margin: 10px 0;
    padding: 4px 10px;
    border-radius: 4px;
    color: $sigWhite;
    background-color: $sigRed;
    font-size: 2rem;
  }
  h2 {
    margin: 0;
    padding: 0;
    color: $sigGrey;
    font-weight: lighter;
    font-size: 1.2rem;
  }
}
