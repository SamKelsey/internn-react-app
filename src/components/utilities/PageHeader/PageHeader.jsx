import React, { Component } from "react";

import "./pageHeader.scss";

class PageHeader extends Component {
  render() {
    const { title, subtitle } = this.props;

    return (
      <div className="pageHeader">
        <h1>{title}</h1>
        {subtitle && <h2>{subtitle}</h2>}
      </div>
    );
  }
}

export default PageHeader;
