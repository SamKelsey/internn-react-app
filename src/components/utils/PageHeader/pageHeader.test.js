import PageHeader from "./PageHeader";
import ShallowRenderer from "react-test-renderer/shallow";

const renderer = new ShallowRenderer();

describe("Component renders correctly", () => {
  test("title and subtitle render correctly", () => {
    renderer.render(<PageHeader title="title" subtitle="subtitle" />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  test("no subtitle rendered if not provided in props", () => {
    renderer.render(<PageHeader title="title" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
