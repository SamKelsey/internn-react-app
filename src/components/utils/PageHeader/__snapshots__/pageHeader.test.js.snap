// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly no subtitle rendered if not provided in props 1`] = `
<div
  className="pageHeader"
>
  <h1>
    title
  </h1>
</div>
`;

exports[`Component renders correctly title and subtitle render correctly 1`] = `
<div
  className="pageHeader"
>
  <h1>
    title
  </h1>
  <h2>
    subtitle
  </h2>
</div>
`;
